app/models/spree/shipment_decorator.rb

```diff
@@ -1,7 +1,13 @@
 module Spree
   Shipment.class_eval do
     def ensure_correct_adjustment_with_included_tax
-      ensure_correct_adjustment_without_included_tax
+      if adjustment
+        ensure_correct_adjustment_without_included_tax
+      elsif selected_shipping_rate_id
+        # ensures the attribute adjustment is set (which, in original spree method, even with the reload is not happening)
+        adjustment = shipping_method.create_adjustment shipping_method.adjustment_label, order, self, true, "open"
```

Comment:
Is this `adjustment` local var really visible in line 11? 🤔

Reply:
adjustment is not a local var, adjustment is a has_one relation in shipment.
That is why the fix works.
This is the shipment method in spree:
https://github.com/openfoodfoundation/spree/blob/2-0-4-stable/core/app/models/spree/shipment.rb#L273

Reply:
ok, great, thanks a lot for the heads up.
You are completely right. I missed the simple fact I am setting the adjustment directly, not a property of it...
The solution would be to use self.adjustment as you suggest.

But, while debugging this and working on the order spec #2823 I realized we dont need this code. The fix will be on #2823 where the order.adjustments association is fixed.

On this PR, I have dropped this commit.